packages/devtools_server/lib/src/server.dart

```diff
@@ -156,24 +156,33 @@ void serveDevTools({
 }
 
 Future<void> registerLaunchDevToolsService(
-  Uri uri,
+  Uri vmServiceUri,
   int id,
   String devToolsUrl,
   bool machineMode,
 ) async {
   try {
     // Connect to the vm service and register a method to launch DevTools in
     // chrome.
-    final VmService service = await _connectToVmService(uri);
+    final VmService service = await _connectToVmService(vmServiceUri);
 
-    service.registerServiceCallback(launchDevToolsService, (request) async {
-      // TODO(kenzie): modify this to append arguments (i.e. theme=dark). This
-      // likely will require passing in args.
-      final url = '$devToolsUrl/?uri=${Uri.encodeComponent(uri.toString())}';
+    service.registerServiceCallback(launchDevToolsService, (params) async {
+      final uriParams = <String, dynamic>{};
+
+      // Copy over queryParams passed by the client
+      if (params != null) {
+        params['queryParams']?.forEach((key, value) => uriParams[key] = value);
+      }
+
+      // Plus the URI to the VM service
```

Comment:
nit: maybe rephrase this comment to be "Add the VM service uri param". I was confused by this wording initially.

Reply:
Done!

It was written as a continuation of the comment above, but that probably wasn't a great idea.

I've also rebased on master, which I think will fix the broken build (I think it was the issue you solved yesterday).